components/branches/branches.js

```diff
@@ -129,17 +129,21 @@ class BranchesViewModel {
         details: 'Deleting ' + details + ' branch cannot be undone with ungit.',
       })
       .show()
-      .closeThen((diag) => {
+      .closeThen(async (diag) => {
         if (!diag.result()) return;
         const url = `${branch.isRemote ? '/remote' : ''}/branches`;
-        return this.server
-          .delPromise(url, {
+
+        try {
+          await this.server.delPromise(url, {
             path: this.graph.repoPath(),
             remote: branch.isRemote ? branch.remote : null,
             name: branch.refName,
-          })
-          .then(() => programEvents.dispatch({ event: 'working-tree-changed' }))
-          .catch((e) => this.server.unhandledRejection(e));
+          });
+
+          return programEvents.dispatch({ event: 'working-tree-changed' });
+        } catch (e) {
+          return this.server.unhandledRejection(e);
```

Comment:
As [`unhandledRejection`](https://github.com/FredrikNoren/ungit/blob/70eb8246421658358d3ef08cfbe6a1974bd9b3b9/public/source/server.js#L180) doesn't seem to return anything, I would remove the `return` here;
```suggestion
          this.server.unhandledRejection(e);
```